Cds 13 dialog fast review

CataldoMazzilli opened the pull request.

CataldoMazzilli committed:
refactor(Modal): modal fast review

refs: CDS-13

CataldoMazzilli committed:
refactor(Modal): modal typescript migration

refs: CDS-13

CataldoMazzilli committed:
Merge remote-tracking branch 'origin/CDS-24-design-system-demo-review' into CDS-13-dialog-fast-review

beawar commented on src/components/feedback/Modal.tsx:
```diff
@@ -80,69 +85,70 @@ export const ModalContainer = styled.div`
 	pointer-events: none;
 `;
-export const ModalContent = styled(Container)`
+const ModalContent = styled(Container)`
```
define props here instead of casting them on usage (size, background)

CataldoMazzilli committed:
refactor(Modal)!: modify props default

BREAKING CHANGE: move showCloseIcon default to true
BREAKING CHANGE: add onClose default (noop)

refs: CDS-13
